### ISSUE TITLE
Revive 2d10

### ISSUE THREAD

**Issue by mlenser** [mlenser/kryx-rpg-issues#516]
I removed it for the reasons specified in https://bitbucket.org/mlenser/tabletop-homebrew/issues/314/solve-2d10s-problems-or-abandon-it, but I believe those can be overcome.

I've long thought that 2d10 is overall a better system. https://faustusnotes.com/2018/05/23/what-is-the-best-basic-die-roll-for-dd-5th-edition-a-probabilistic-analysis/ covers some of the math

If I go through with this I'll make 2d10 the default, but keep 1d20 as a setting.

**Comment by itamarcu** [comment]
I think it's wrong to discard this idea because of numbers being a bit above or a bit below the RAW numbers - it looks to me like this can be easily fixed by just bumping proficiency modifiers or DCs or Defense by 1 or 2 (up or down). 

However, I think there needs to be ***a lot more*** discussion about the impact of the *shape* of this curve. The article linked in https://faustusnotes.com spends 95% of the text just going through charts, and only barely mentions the actual pros and cons of this change, which I dislike - D&D (especially 5e) is heavily designed according to what *feels* good.

At the most basic level, this change would mean that easy things are easier and harder things are harder. The "common" probability of ~60% success at an average task (such as stabbing or shooting an enemy) may remain the same, but the extremities skew further up and down, making everything more consistent in a way that isn't always more fun.

I know a lot of people want skill proficiencies to be more consistent, and while this does it, it isn't necessarily the best solution, as it has a lot of side effects. 

I'm strongly in favor of listing exactly the problems this means to solve, and listing the potential problems this can create (for example, I think it'll really hurt the idea of bounded accuracy for battles against very strong or very weak enemies).

I'm less interested in discussions about numbers for - those can be tweaked rather easily. For example, if using 2d10 ends up causing a lot more near-misses (where half damage is dealt), this is not a real problem because it can be fixed by reducing the miss threshold from 4 to 3. Similarly, the problem of critical chances changing from 5% to 1% can be fixed by redefining them to be "roll a 19+" or "roll 9+ on both dice".

**Comment by mlenser** [comment]
I agree with what you're saying. It should definitely be fully explained and explored before I implement it. I wrote about the numbers because the math was the main reason it was removed before.

I'll write pros/cons later

**Comment by mlenser** [comment]
The goal of this change is that a character attempting a task (picking a lock, swinging a sword, remembering their history, jumping across a gap, etc) will generally have consistent results.

A jumper will generally jump ~10-15 feet (generic numbers, not to be taken as statistical math), pretty consistently. There could some some cases of a jump that is perfect and the result is longer or some cases that the character stumbles and the jump is much shorter, but I believe those cases should be much less common.

Ignoring success rates, math, and mechanics, more consistent results is the goal.

I believe 2d10 could be implemented currently without the game having very many cracks. Though we could probably refine a few things.

____________________

Math:
![image](https://user-images.githubusercontent.com/235857/91632613-4fdd3800-e9e2-11ea-870c-a9d02e5ed4b1.png)
![image](https://user-images.githubusercontent.com/235857/91632616-54a1ec00-e9e2-11ea-83cb-428614905858.png)

**Comment by itamarcu** [comment]
A lot of these changes would be great - especially for skills (like jumping or picking a lock).

For fighting ("swinging a sword"), it's a completely different issue. An expert athlete will certainly jump perfectly almost every time, but fighting is inherently a lot more random. If a system like this is applied to combat, it will inevitably break the idea of low-level threats still being a danger.

With the current system, a CR 1/4 Goblin will hit a level 10 character about 30% of the time, and a CR 15 Mummy Lord will hit about 60% of the time. If we use a system like 2d10, these chances will have to change to something like 5% and something like 90%. The chances for the players to hit the monsters will be roughly the same and flipped, of course.

I'm not immediately opposed to this, but this will bring us back to older D&D systems, where low level threats are not a problem. 

If the goal is just to solve consistency out of combat, I think there are other solutions that can be implemented.

For example, making passive skills a "floor" for active skills, or adding mechanics that only care about your modifier and don't require your roll.

There could also be meta-mechanics like Inspiration which allow the players to get consistency in moments that matter, but not always. There's a lot of tabletop RPGs (and some [video games](https://store.steampowered.com/app/527230/For_The_King/)) where the players can spend "inspiration", "focus", "fate points", "essence", etc. for D&D, these could allow you to "take 15" (or 10 or 19) instead of rolling. alternatively, they could give you a bonus after you've made your roll, much like bardic inspiration.

So, in summary - I think there can be other solutions to make skill checks consistent (or even a partial solution: use 2d10 for skills and 1d20 for combat). I don't see a way where combat could be changed to 2d10 without making extremely strong/weak enemies even more/less of a threat. In other words - some monsters will be consistently trivial or consistently unbeatable.

**Comment by mlenser** [comment]
> For fighting ("swinging a sword"), it's a completely different issue.

I don't think it is. A goblin can shove just as much as it can swing its scimitar. While both sides of a shove are rolled, the difference between bonuses is still amplified.

> If a system like this is applied to combat, it will inevitably break the idea of low-level threats still being a danger.
> With the current system, a CR 1/4 Goblin will hit a level 10 character about 30% of the time

I don't believe it would. I'm not seeing how you concluded your numbers. Anydice:
### Level 1-4 (Defense ~16)
- `output 1d20 + 4 > 16`:  40%
- `output 2d10 + 4 > 16`:  36%

### Level 1-4 with shield or Level 5+ (Defense ~18)
- `output 1d20 + 4 > 18`:  30%
- `output 2d10 + 4 > 18`:  21%

### Level 5+ with shield (Defense ~20)
- `output 1d20 + 4 > 20`:  20%
- `output 2d10 + 4 > 20`:  10%

**Comment by Lamorak11** [comment]
I'll be frank, I have nothing to add about the validity of a 2d10 system. 

However, I will say, for the sake of my position being out there and in case there are others that agree, that the d20 is a sacred cow that I am flat-out unwilling to give up.

If, as was stated, the d20 system would remain fully usable at the same time as a 2d10 system, then I have no issues with the addition. However, if there is a point where the d20 is no longer supported, my play group and I will regretfully have to go back to vanilla. As I am extremely invested in KRYXRPG, I really don't want this, but hopefully a solution can be found that avoids this. 

I am stating this here as I didn't want this discussion to get to an advanced point without at least pointing out my opposition.

**Comment by mlenser** [comment]
@Lamorak11 as I wrote in the OP:
> If I go through with this I'll make 2d10 the default, but keep 1d20 as a setting.

**Comment by Paulorpribeiro** [comment]
The main con I see with 2d10 relative to 1d20 is that the +1 mod does not mean a 5% increase anymore, but actually range between 2% (if you needed a 20 and now need a 19) and 10% (if you needed 11 and now need a 10). This means that a maneuver or spell that gave +2 defense has to be rebalance. Not only those, but all spells, maneuvers and class features that give bonus/penalties to attack, defense, saves or skills will have to at rebalance or at least checked if they are not too powerfull or too weak. I think that is a massive work for the benefit it brings

**Comment by itamarcu** [comment]
@Paulorpribeiro - I think many of the pieces of the system that give ± to attacks, defence, etc - are balanced around affecting a "standard" chance (of around 55% to hit), and do not easily stack to high numbers. The system purposefully makes it difficult to stack multiple such things together, and the chances of success tend to consistently hover around that range, so I don't think there are such potential problems that aren't already problems in the existing system.

By the way, another thing came to mind - with this chance, the chance for "near misses" increases from 20% to 30% (assuming 10+ is a hit).

**Comment by Paulorpribeiro** [comment]
I see, this change will only exacerbate (is that a word? Enhance might be better) the problems already in place. For exanple, getting a +2 defense as a reaction when the enemy already hits you only on a 18+ decreases his chances from 6% to 1%. Your maneuver had the value of +1 comparing to the old system. On the other hand if the enemy needed a 10 to hit you, his chances decreased from 64% to 55%, almost the 10% it was in the old system.

Another thing to consider, that I just figured out, is that advantage is worth less with 2d10 than it did with 1d20.
In 1d20 it was worth between +5 and +2, higher on the mid range. With 2d10 it varies between +1 and +3, higher in the mid range. This can also be adjusted, but one more detail to keep in mind

**Comment by Marcloure** [comment]
My main concern about the bonuses stopping being constant is that the players and GMs (unless they have good statistical knowledge) will not know if it's better to use shield of faith on the plate and shield paladin or on the greataxe berserker. They may use it on one target unknowing that the bonus on it will be less effective than they expect.

The flat d20 doesn't have this issue, a +2 is equally valuable to both characters, so there isn't less effective use.

**Comment by mlenser** [comment]
As itamarcu allued: we can fix the numbers. Statistical outcomes of small number bonuses can be adjusted. And if we decide to go through with this, I can outline some examples and how the math changes. Cases like Shield of Faith, Sharpshooting Fighting Style, Shield, Partial hits, etc will have a statistical value similar to what they are today. That's not difficult or much effort.

So lets please ignore that for now.
______________

One of the biggest issues I'm working through in my head right now is regardign defense and keeping lower challenge creatures a threat while not making defense nearly worthless against threats around the same challenge as you progress in levels.

I'll have to fix that for both 2d10 and 1d20, so maybe best to fix that now.

**Comment by mlenser** [comment]
## Regarding ability checks and 2d10.

[see Skills math](https://docs.google.com/spreadsheets/d/1sGvcz9vbq9yk4kRBEY6ZYGWhGO6-t7CWudznQcWPk74/edit#gid=444002489)

vs a Medium DC
1d20: 85% for prof, 70% for no prof
2d10: 94% for prof, 85% for no prof
Expertise was 95%

vs a Hard DC
1d20: 60% for prof, 45% for no prof
2d10: 72% for prof, 45% for no prof
Expertise was 75%

vs a Very Hard DC
1d20: 25% for prof, 20% for no prof
2d10: 28% for prof, 10% for no prof
Expertise was 50%

So Medium DCs are ok and Very Hard DCs are not meaningful changed (3% difference). So the end result here is that 2d10 is no where near as bad as expertise was for Medium DCs or especially Very Hard DCs. 2d10 makes Hard checks easier for those with proficiency (one could argue this is problematic), but does not seemingly have the same problems with even harder DCs.

For me this is enough evidence that I feel comfortable with 2d10 for skills. It allows those with specialization the ability to have more reliability without causing the major problems of "all or nothing".

**Comment by mlenser** [comment]
## Defenses and saves

See the math (a bit unorganized): https://docs.google.com/spreadsheets/d/1sGvcz9vbq9yk4kRBEY6ZYGWhGO6-t7CWudznQcWPk74/edit#gid=158786285

2d10 and partial hits allows low challenge monsters to continue to be threats into high levels far better than RAW does.

### Compared to 1d20 & partial hits
- Plate & shield damage is ~90% of what it is with 1d20
- Plate damage is ~109% of what it is with 1d20
- Scale, Hide, Leather, and Mage Armor damage is ~119% of what it is with 1d20

It is reasonable that Plate & shield is extraordinarily tanky.

### Compared to 1d20 without partial hits
- Plate & shield damage is ~145% of what it is with 1d20
- Plate damage is ~156% of what it is with 1d20
- Scale, Hide, Leather, and Mage Armor damage is ~159% of what it is with 1d20

**Comment by mlenser** [comment]
The problems of 2d10 that [caused me to abandon it](https://bitbucket.org/mlenser/tabletop-homebrew/issues/314/solve-2d10s-problems-or-abandon-it) are not a problem with partial hits and partial sucesses.

**Comment by mlenser** [comment]
2d10 added as an alternative to 1d20 and set as the default option
